docs/architecture.md

```diff
@@ -0,0 +1,23 @@
+# Packages
+
+All packages follow Gatsby's [plugin naming scheme](https://www.gatsbyjs.com/docs/plugins-themes-and-starters/). 
+
+* **gatsby-plugin-graphql** adds Webpack and Babel configuration so we can share queries and types between Gatsby and any other GraphQL server. This plugin is essential for dynamic, client side queries
+
+* **gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, this plugins adds all necessary providers to the React tree
```

Comment:
This doesn't mean that dynamic interpolations would not happen right?

I mean, if I have a component that renders something like this based on the cart number of items:

> There are 3 products in your cart.

or

> There are 1 product in your cart.

or

> There are no products in your cart.

This should be handled by an i18n library in the client

Reply:
No, we still have react-intl, however, we implement [this guide](https://formatjs.io/docs/guides/advanced-usage#pre-compiling-messages) to reduce react-intl sizes and blocking. time

Reply:
Awesome!